### ISSUE TITLE
[Bug]: Attributes are not being preserved with temporal averaging APIs using `xarray > 2023.3.0`

### ISSUE THREAD

**Issue by bonfils2** [xCDAT/xcdat#458]
### What happened?

In the xcdat dependencies, xarray version is set to 2023.3.0, but it has been updated to newer version when you were installing new jupyter notebook in your env.

### What did you expect to happen?

I expected the code to run, as usual. 

### Minimal Complete Verifiable Example

```python
import xcdat as xc
ds = xc.open_dataset('/p/css03/esgf_publish/CMIP6/ScenarioMIP/IPSL/IPSL-CM6A-LR/ssp585/r2i1p1f1/Amon/tas/gr/v20191121/tas_Amon_IPSL-CM6A-LR_ssp585_r2i1p1f1_gr_201501-210012.nc', center_times=True)
ds = ds.temporal.group_average("tas", freq="year")
weights = ds.spatial.get_weights(axis=['Y', 'X'], lat_bounds=[-90,90],lon_bounds=[0,360])
```

### Relevant log output

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lee1043/git/xcdat/xcdat/spatial.py", line 274, in get_weights
    d_bounds = self._dataset.bounds.get_bounds(axis=key, var_key=data_var)
  File "/home/lee1043/git/xcdat/xcdat/bounds.py", line 243, in get_bounds
    raise KeyError(
KeyError: "No bounds data variables were found for the 'Y' axis. Make sure the dataset has bound data vars and their names match the 'bounds' attributes found on their related time coordinate variables. Alternatively, you can add bounds with `xcdat.add_missing_bounds` or `xcdat.add_bounds`."
```

### Anything else we need to know?

The problem was fixed with : conda install xarray=2023.3.0

### Environment

0.5.0
2023.4.2

**Comment by tomvothecoder** [comment]
This issue is related to a line in the temporal averaging API for generating weights, not spatial averaging.

https://github.com/xCDAT/xcdat/blob/cbef082d9f3ea43aa7e4e36f4d2b23877a8826bc/xcdat/temporal.py#L1141

In the latest version of xarray, `xr.where()` with `np.isnan(dv)` deletes the attributes on `dv`. The fix is to replace `np.isnan(dv)` with `dv.copy().isnull()`. Not exactly sure what changed in xarray.

**Comment by lee1043** [comment]
Related discussion: https://github.com/xCDAT/xcdat/discussions/455 (leaving this for record)

**Comment by tomvothecoder** [comment]
Hey @bonfils2, just a heads up that I released a patch for v0.5.0 on [conda-forge](https://anaconda.org/conda-forge/xcdat/files) to fix the issue of attributes being dropped when calculating weighted temporal averages.

conda/mamba should automatically pull the latest version. If you want to update xcdat in an existing environment, you can run `mamba update -c conda-forge xcdat` or `mamba install -c conda-forge xcdat=0.5.0=pyhd8ed1ab_1`.
